augur/ancestral.py

```diff
@@ -323,8 +323,12 @@ def run(args):
         from .utils import load_features
         ## load features; only requested features if genes given
         features = load_features(args.annotation, args.genes)
-        if features is None:
-            raise AugurError("could not read features of reference sequence file")
+        # Ensure the already-created nuc annotation coordinates match those parsed from the reference file
+        if (features['nuc'].location.start+1 != anc_seqs['annotations']['nuc']['start'] or
+            features['nuc'].location.end != anc_seqs['annotations']['nuc']['end']):
+            raise AugurError(f"The 'nuc' annotation coordinates parsed from {args.annotation!r} ({features['nuc'].location.start+1}..{features['nuc'].location.end})"
```

Comment:
It's a bummer that this `+ 1` logic needs to exist in so many places (ancestral, translate, and utils) to communicate or compare coordinates between different systems. It's the kind of logic that is easy to forget to include in future code and then lead to subtle bugs. I don't have a great suggestion to fix this except maybe encapsulating or replacing the `SeqFeature` functionality so we can interact with objects that format coordinates in our expected "one-origin, inclusive" format. That's outside of the scope of this PR though.

Reply:
Totally. Short of adding another layer on top of `SeqFeature` (which might alleviate this problem but at the cost of another layer to understand) I've leaned heavily on writing tests to catch this. Auspice has this issue as well and [tests](https://github.com/nextstrain/auspice/blob/master/test/annotation-parsing.test.js) were the only path to sanity!

Reply:
Oh - so I believe you can monkeypatch things in Python. For instance, adding

```py
from Bio.SeqFeature import FeatureLocation
def start_one_based(self):
    return self.start + 1
FeatureLocation.start_one_based = start_one_based
```

at the top of `utils.py` seemingly allows us to call `<feature>.location.start_one_based`, even when those features are created by `SeqIO.read(...)`, and even if we re-import `FeatureLocation`. I'd need to read more about class inheritance and import resolution before I'd be confident doing this, so I don't want to do it in this PR, but it seems like it's posisble.